lib/msgr/connection.rb

```diff
@@ -38,12 +37,16 @@ def connect
       connection
     end
 
-    def channel
-      @channel ||= begin
-        channel = connection.create_channel
-        channel.prefetch 1
-        channel
-      end
+    def channel(prefetch = 1)
+      raw_channel = connection.create_channel
+      raw_channel.prefetch prefetch
+      channel = Msgr::Channel.new(config, raw_channel)
+      @channels << channel
+      channel
```

Comment:
Have you thought about moving `connection.create_channel` and setting prefetch into `Msgr::Channel`. Like in `.new` or `#initialize`. This might make this method smaller and keep the channel logic at one place.